model/src/main/kotlin/utils/SimplePackageConfigurationProvider.kt

```diff
@@ -43,7 +45,15 @@ class SimplePackageConfigurationProvider(
         fun forDirectory(directory: File): SimplePackageConfigurationProvider {
             val entries = directory.walkBottomUp()
                 .filterTo(mutableListOf()) { !it.isHidden && it.isFile }
-                .map { it.readValue<PackageConfiguration>() }
+                .map { file ->
+                    try {
+                        file.readValue<PackageConfiguration>()
+                    } catch (e: IOException) {
+                        log.error { "Error reading package configuration from '${file.absoluteFile}'." }
+
+                        throw e
```

Comment:
Wouldn't it be better to just wrap the exception here with another exception that contains above message and let the calling code decide how to handle this? Because otherwise the log message above could be separated from the other related logs and difficult to associate.

Reply:
The log I tested it with had ~10 lines callstack and right above it the log message which I found quite discoverable. I don't see an issue with that, can we keep it?

Reply:
I'd still find it cleaner to wrap the exception instead. Otherwise this should follow our convention to add `": ${e.collectMessagesAsString()}"` to the message and also call `e.showStackTrace()`.

Reply:
How would you wrap it?

Reply:
however, I updated the change as you desired.